src/chain/chain.rs

```diff
@@ -1484,9 +1473,9 @@ impl Chain {
 #[derive(Default)]
 pub struct EldersChange {
     // Peers that became elders.
-    pub added: BTreeSet<P2pNode>,
+    pub added: BTreeMap<PublicId, P2pNode>,
     // Peers that ceased to be elders.
-    pub removed: BTreeSet<P2pNode>,
+    pub removed: BTreeMap<PublicId, P2pNode>,
```

Comment:
I don't think this change was necessary. We use this stuct only to update the neighbour connections (`Elder::update_neighbour_connections`) and we don't need to lookup by `PublicId` there.

Reply:
Yes with the new implementation of `Ord` of `P2pNode` we can probable go back to having `BTreeSet<P2pNode>`, without any wrappers even probably?

Reply:
Reverted to `BTreeSet<P2pNode>` for Elder updates in the latest revision